File: Desktop/Domain/CmdMsgs/CreateContactCmd.cs
using System;

using System.Xml;
using System.Xml.Xsl;
using System.Collections;
using System.Runtime.InteropServices;

namespace Blurts
{
  [ComVisible(true)]
  public class CreateContactCmd : CmdBase
  {
    // Tag Names
    private static String KEY_PREFIXNAME = "PrefixName";
    private static String KEY_FIRSTNAME = "FirstName";
    private static String KEY_LASTNAME = "LastName";
    private static String KEY_WORK = "PhoneWork";
    private static String KEY_MOBILE = "PhoneMobile";
    private static String KEY_PHOTO = "Photo";
    private static String KEY_ORG = "Org";
    private static String KEY_PIN = "Pin";
    private static String KEY_EMAIL = "Email";


    public CreateContactCmd(String s)
      : base(s)
    {
    }

    public CreateContactCmd()
      : base(CMD_CREATECONTACT)
    {
    }

    [ComVisible(true)]
    public String PrefixName
    {
      get
      {
        return optString(KEY_PREFIXNAME);
      }
      set
      {
        try
        {
          remove(KEY_PREFIXNAME);
          put(KEY_PREFIXNAME, value);
        }
        catch (JSONException e)
        {
          Console.WriteLine(e.ToString());
        }
      }
    }

    public String FirstName
    {
      get
      {
        return optString(KEY_FIRSTNAME);
      }
      set
      {
        try
        {
          remove(KEY_FIRSTNAME);
          put(KEY_FIRSTNAME, value);
        }
        catch (JSONException e)
        {
          Console.WriteLine(e.ToString());
        }
      }
    }

    public String LastName
    {
      get
      {
        return optString(KEY_LASTNAME);
      }
      set
      {
        try
        {
          remove(KEY_LASTNAME);
          put(KEY_LASTNAME, value);
        }
        catch (JSONException e)
        {
          Console.WriteLine(e.ToString());
        }
      }
    }

    public String PhoneWork
    {
      get
      {
        return optString(KEY_WORK);
      }
      set
      {
        try
        {
          remove(KEY_WORK);
          put(KEY_WORK, value);
        }
        catch (JSONException e)
        {
          Console.WriteLine(e.ToString());
        }
      }
    }

    public String PhoneMobile
    {
      get
      {
        return optString(KEY_MOBILE);
      }
      set
      {
        try
        {
          remove(KEY_MOBILE);
          put(KEY_MOBILE, value);
        }
        catch (JSONException e)
        {
          Console.WriteLine(e.ToString());
        }
      }
    }


    [ComVisible(true)]
    public String Photo
    {
      get
      {
        return optString(KEY_PHOTO);
      }
      set
      {
        try
        {
          remove(KEY_PHOTO);
          put(KEY_PHOTO, value);
        }
        catch (JSONException e)
        {
          Console.WriteLine(e.ToString());
        }
      }
    }

    [ComVisible(true)]
    public String Org
    {
      get
      {
        return optString(KEY_ORG);
      }
      set
      {
        try
        {
          remove(KEY_ORG);
          put(KEY_ORG, value);
        }
        catch (JSONException e)
        {
          Console.WriteLine(e.ToString());
        }
      }
    }

    [ComVisible(true)]
    public String Pin
    {
      get
      {
        return optString(KEY_PIN);
      }
      set
      {
        try
        {
          remove(KEY_PIN);
          put(KEY_PIN, value);
        }
        catch (JSONException e)
        {
          Console.WriteLine(e.ToString());
        }
      }
    }

    [ComVisible(true)]
    public String Email
    {
      get
      {
        return optString(KEY_EMAIL);
      }
      set
      {
        try
        {
          remove(KEY_EMAIL);
          put(KEY_EMAIL, value);
        }
        catch (JSONException e)
        {
          Console.WriteLine(e.ToString());
        }
      }
    }
  }
}

File: BlackBerry/AlertMsgs/CreateContactCmd.java
package com.mlhsoftware.Blurts;

import org.json.me.*;

import net.rim.blackberry.api.pdap.*;
import javax.microedition.pim.*;
import net.rim.device.api.system.*;
import java.util.Vector;

class CreateContactCmd extends CmdBase
{
  // Tag Names
  private static final String ALERT_NAME = "Contact";


  private static final String KEY_UID = "Uid";
  private static final String KEY_PREFIXNAME = "PrefixName";
  private static final String KEY_FIRSTNAME = "FirstName";
  private static final String KEY_LASTNAME = "LastName";
  private static final String KEY_DISPLAYNAME = "DisplayName";
  private static final String KEY_PHOTO = "Photo";
  private static final String KEY_ORG = "Org";
  private static final String KEY_PIN = "Pin";
  private static final String KEY_EMAIL = "Emails";
  private static final String KEY_TEL = "PhoneNumbers";
  private static final String KEY_WORK = "PhoneWork";
  private static final String KEY_MOBILE = "PhoneMobile";




  public CreateContactCmd( String string ) throws JSONException
  {
    super( string );
  }


  public void setUID( String value )
  {
    try
    {
      put( KEY_UID, value );
    }
    catch ( JSONException e )
    {
      System.out.println( e.toString() );
    }
  }

  String getUID()
  {
    return optString( KEY_UID );
  }

  public void setPrefixName( String value )
  {
    try
    {
      put( KEY_PREFIXNAME, value );
    }
    catch ( JSONException e )
    {
      System.out.println( e.toString() );
    }
  }

  String getPrefixName()
  {
    return optString( KEY_PREFIXNAME );
  }

  public void setFirstName( String value )
  {
    try
    {
      put( KEY_FIRSTNAME, value );
    }
    catch ( JSONException e )
    {
      System.out.println( e.toString() );
    }
  }

  String getFirstName()
  {
    return optString( KEY_FIRSTNAME );
  }

  public void setLastName( String value )
  {
    try
    {
      put( KEY_LASTNAME, value );
    }
    catch ( JSONException e )
    {
      System.out.println( e.toString() );
    }
  }

  String getLastName()
  {
    return optString( KEY_LASTNAME );
  }

  public void setDisplayName( String value )
  {
    try
    {
      put( KEY_DISPLAYNAME, value );
    }
    catch ( JSONException e )
    {
      System.out.println( e.toString() );
    }
  }

  String getDisplayName()
  {
    return optString( KEY_DISPLAYNAME );
  }


  public void setPhoto( String value )
  {
    try
    {
      put( KEY_PHOTO, value );
    }
    catch ( JSONException e )
    {
      System.out.println( e.toString() );
    }
  }

  String getPhoto()
  {
    return optString( KEY_PHOTO );
  }


  public void setPin( String value )
  {
    try
    {
      put( KEY_PIN, value );
    }
    catch ( JSONException e )
    {
      System.out.println( e.toString() );
    }
  }

  String getPin()
  {
    return optString( KEY_PIN );
  }


  public void setOrg( String value )
  {
    try
    {
      put( KEY_ORG, value );
    }
    catch ( JSONException e )
    {
      System.out.println( e.toString() );
    }
  }

  String getOrg()
  {
    return optString( KEY_ORG );
  }

  String getWorkNumber()
  {
    return optString( KEY_WORK );
  }

  String getMobileNumber()
  {
    return optString( KEY_MOBILE );
  }

  String getEmail()
  {
    return optString( KEY_EMAIL );
  }
}